### DISCUSSION TITLE
Download error

### DISCUSSION THREAD

**Discussion by Budhi108** [aclap-dev/video-downloadhelper#636] [Help and Q&A]
### Your question

You ask me to send report:

Sadly we failed at downloading that specific media. We try to support as many websites as possible, so it would help a lot if you could report that error!

{
  "version": "9.0.1.18",
  "target": "google",
  "channel": "stable",
  "page_url": "undef",
  "error": {
    "details": "SideDownload error: : [tls @ 00000201e20e4040] IO error: Error number -10054 occurred\r\n    Last message repeated 1 times\r\n[aac_adtstoasc @ 00000201e20ccf40] Error parsing ADTS frame header!\r\n[mp4 @ 00000201e226e600] Error applying bitstream filters to an output packet for stream #1: Invalid data found when processing input\r\n[aost#0:1/copy @ 00000201e2119b00] Error submitting a packet to the muxer: Invalid data found when processing input\r\n[out#0/mp4 @ 00000201e2109400] Error muxing a packet\r\n",
    "downloadable_id": "downloadable_905035407567510",
    "error": "coapp_failure",
    "id": "coapp_failure_a84b0e40-d2eb-4466-8056-bd2c53df457a"
  }
}

### Addon & CoApp version and details

{
  "version": "9.0.1.18",
  "target": "google",
  "channel": "stable",
  "page_url": "undef",
  "error": {
    "details": "SideDownload error: : [tls @ 00000201e20e4040] IO error: Error number -10054 occurred\r\n    Last message repeated 1 times\r\n[aac_adtstoasc @ 00000201e20ccf40] Error parsing ADTS frame header!\r\n[mp4 @ 00000201e226e600] Error applying bitstream filters to an output packet for stream #1: Invalid data found when processing input\r\n[aost#0:1/copy @ 00000201e2119b00] Error submitting a packet to the muxer: Invalid data found when processing input\r\n[out#0/mp4 @ 00000201e2109400] Error muxing a packet\r\n",
    "downloadable_id": "downloadable_905035407567510",
    "error": "coapp_failure",
    "id": "coapp_failure_a84b0e40-d2eb-4466-8056-bd2c53df457a"
  }
}

**Comment by GrampaWildWilly** [comment]
> "page_url": "undef",

That's extremely weird.  That is supposed to contain a URL.  Anybody who would try to help you needs that URL.  Oh well.  So the automatic problem reporting mechanism didn't report this one critical piece of information.  Not a problem.  Do it yourself.  Copy/paste into a post here the URL of the web page on which your problematic video resides & let's see if we can make some progress.

**Reply by Budhi108** [reply]
Thanks for answer

here is link:

https://turskeserije.tv/yali-capkini-epizoda-71/

I try download many times...and everytime download stops on 82% or 93%  or so...and get error. Yesterday on old version was everything ok

I must say 480p version was downloaded ok but 720 version always give error

**Comment by Budhi108** [comment]
and again:

{
  "version": "9.0.1.18",
  "target": "google",
  "channel": "stable",
  "page_url": "https://turskeserije.tv/yali-capkini-epizoda-71/",
  "error": {
    "details": "SideDownload error: : [tls @ 0000019b0261a180] IO error: Error number -10054 occurred\r\n    Last message repeated 1 times\r\n[tls @ 0000019b04354040] IO error: Error number -10054 occurred\r\n    Last message repeated 1 times\r\n[tls @ 0000019b04d12940] IO error: Error number -10054 occurred\r\n    Last message repeated 1 times\r\n[tls @ 0000019b042f0340] IO error: Error number -10054 occurred\r\n    Last message repeated 1 times\r\n[aac_adtstoasc @ 0000019b0433cc40] Error parsing ADTS frame header!\r\n[mp4 @ 0000019b0436a500] Error applying bitstream filters to an output packet for stream #1: Invalid data found when processing input\r\n[aost#0:1/copy @ 0000019b04382940] Error submitting a packet to the muxer: Invalid data found when processing input\r\n[out#0/mp4 @ 0000019b04379640] Error muxing a packet\r\n",
    "downloadable_id": "downloadable_3503664939270414",
    "error": "coapp_failure",
    "id": "coapp_failure_4e3737bc-5905-48e0-bb6f-a8c00bd2cdf4"
  }
}

**Comment by Budhi108** [comment]
Hi there,

so far I have never had such a problem with the old version of the application, so maybe it will help you to find out what is wrong with the new version. Now, I downloading some episodes a couple days and everything was ok. I try same episodes download now and there an issue, so new version of app probably 
give a problem

best regards

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.22a1 beta, CoApp 2.0.19.

I visited your movie & got this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/b536540a-7264-4dea-a87a-b086ed1fc382)

The download was progressing, although at a speed that probably would have taken 8 hours to complete, when the download terminated with these errors:

SideDownload error: : [tls @ 0000027b07320340] IO error: Error number -10054 occurred
    Last message repeated 5 times
[tls @ 0000027b06f6a100] IO error: Error number -10054 occurred
    Last message repeated 1 times
[tls @ 0000027b06ef3b40] IO error: Error number -10054 occurred
    Last message repeated 1 times
[aac_adtstoasc @ 0000027b07371ec0] Error parsing ADTS frame header!
[mp4 @ 0000027b07da6d00] Error applying bitstream filters to an output packet for stream #1: Invalid data found when processing input
[aost#0:1/copy @ 0000027b0739ae40] Error submitting a packet to the muxer: Invalid data found when processing input
[out#0/mp4 @ 0000027b073b7140] Error muxing a packet

I did have an MP4 that I could play, but it was only 2 minutes long, barely enough for the opening credits.  This movie is, according to ffprobe, 2 hours & 24 minutes long.

This is most disturbing.

So I went to Plan B.  You can see an HLS master manifest in my image above.  I ran its URL through ffprobe & got these results:

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/15443293/ffprobe.txt)

There are just 2 resolutions on offer here.  In addition, there is Timed_ID3 data.  This is junk & should always be ignored.  I believe VDH indeed does ignore it now.  In the past, it used to be a problem, so I always point it out when I encounter it.

Anyway, I used the information from ffprobe to launch a download of the movie via ffmpeg.  It was running at a pretty slow pace as well.  Here is the log from that attempt.

[Turkish Movie mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/15443307/Turkish.Movie.mp4.Log.txt)

You can see that there was about as much corrupt content as there was usable content.  With that much corrupt input, ffmpeg was probably spending most of its time in error correction.  I imagine that's why the download was progressing at such a slow speed.  But this eventually failed as well.  It failed because there was something in the stream that ffmpeg simply could not repair or ignore.  The errors that ffmpeg saw were quite similar to what VDH saw earlier.

Ffmpeg managed to download almost 14 minutes of this.  I suppose that's better than 2 minutes.  It played in VLC but there were problems.  The audio track seemed to be badly out of synch with the video.  And then the audio simply stopped several minutes before the video did.  At least there are burned-in subtitles, although I'm guessing they're in Turkish, which doesn't help me any seeing as it's a language I don't speak.  There were also several places where the video track just froze for a couple of seconds before continuing.  I'm guessing that's evidence of some of the corruption that ffmpeg reported.

The problem with this is not VDH.  The problem is the content itself.  It is of extremely low quality & appears to be damaged.  If you can manage to play this in the web page, I would recommend you try to record it with a screen recorder.  That will take the full 2 hours & 24 minutes of the movie.  But it's better than nothing.  Downloading appears to be a forlorn hope.

**Reply by GrampaWildWilly** [reply]
There is another problem in this forum that I worked on yesterday.  It dealt with Pornhub.  When we first tried to download something, it failed miserably, repeatedly.  A few hours later, everything worked fine.  #588  It is possible we're looking at a similar problem here.  You say you have had no trouble with other content on this site.  Wait until tomorrow.  See if VDH can download this movie then.  I still say this has nothing to do with the version of VDH & everything to do with the content being fed to us by the web site.  We can only hope that things will be better tomorrow.  Normally, I would say such advice is a desperate attempt by somebody who doesn't know what he's doing to put you off.  But that experience with Pornhub makes me suggest the same thing might be happening here.  Let's test my hypothesis & see.  Let's wait until tomorrow.

**Reply by Budhi108** [reply]
Thank you for your trouble. I appreciate it. You are probably right that it is about the video. The video was uploaded a few hours earlier before trying to download it (I don't know if that has anything to do with it?) now I tried downloading the 480p and 720p versions again. downloaded the smaller version in 1 minute (1.1 GB) and the 720 version in about 1.5 minutes (1.63 GB) and everything works as it should.
